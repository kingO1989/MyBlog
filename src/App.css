* {
}

html,
body {
  margin: 0;
  background-color: rgba(245, 245, 245, 0.89);
}

header {
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 40px;
  height: 40px;
  padding: 10px;
}

header .logo_search {
  display: flex;
}
header .logo_search .search input {
  background-color: rgba(245, 245, 245, 0.89);
  border: 1px solid rgba(245, 245, 245, 0.89);
  color: rgba(0, 0, 0, 0.7);
  border-radius: 15px;
  margin-left: 10px;
  height: 20px;
}

header .my_works_social_media_notifications {
  width: 13%;
  display: flex;
  justify-content: space-around;
}
header .my_works_social_media_notifications .fontawesome_item {
  color: black;
  font-size: x-large;
}

header .my_works_social_media_notifications .profile {
  border: 1px solid black;
  border-radius: 200px;
  padding: 5px;
  width: 20px;
  height: 20px;
}

main {
  min-height: 400px;
  display: flex;
  padding: 10px;
}
main .authors_intro {
  width: 68%;
  background-color: white;
  padding-left: 5px;
  padding-right: 5px;
}
main .authors_intro nav {
}
main .authors_intro nav.post_nav {
  background-color: white;
  padding-top: 30px;

  margin-bottom: 30px;
  border-bottom: 1px solid silver;
  display: flex;
  justify-content: space-around;
  z-index: 9;
}

main .authors_intro nav.post_nav .navitem {
  padding-bottom: 10px;
}

main .authors_intro nav.post_nav .navitem:hover {
  cursor: pointer;
}

main .authors_intro nav.post_nav .navitem.active {
  border-bottom: 1px solid;
}
/* Page content */
.content {
  padding: 16px;
}

/* The sticky class is added to the navbar with JS when it reaches its scroll position */
.sticky {
  position: fixed;
  top: 0;
  width: 100%;
}

/* Add some top padding to the page content to prevent sudden quick movement (as the navigation bar gets a new position at the top of the page (position:fixed and top:0) */
.sticky + .allposts {
  padding-top: 60px;
}

main .authors_intro .allposts {
  border: 0px solid red;
}
main .authors_intro .allposts.dark {
  background-color: black;
  color: white;
}

main .authors_intro .allposts .flex-container {
  display: flex;
  flex-direction: column;
}
main .authors_intro .allposts .post_vignette {
  display: flex;
  border-bottom: 1px solid rgba(245, 245, 245, 0.89);
  padding-bottom: 30px;
  margin-bottom: 20px;
}

main .authors_intro .allposts .post_vignette:hover {
  cursor: pointer;
}

main .authors_intro .allposts .post_vignette .post_data {
  width: 70%;
}

main .authors_intro .allposts .post_vignette .post_data .date {
  color: rgba(0, 0, 0, 0.7);
}

main .authors_intro .allposts .post_vignette .post_data .title {
}

main .authors_intro .allposts .post_vignette .post_data .details {
  color: rgba(0, 0, 0, 0.9);
}

main .authors_intro .allposts .post_vignette .post_data .vignette_footer {
  display: flex;
  justify-content: space-between;
}

main
  .authors_intro
  .allposts
  .post_vignette
  .post_data
  .vignette_footer
  .tags_and_time_to_read {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

main
  .authors_intro
  .allposts
  .post_vignette
  .post_data
  .vignette_footer
  .tags_and_time_to_read
  .tag {
  background-color: rgb(245, 243, 243);

  border-radius: 9px;
  padding: 5px;
  margin-left: 5px;
}
main
  .authors_intro
  .allposts
  .post_vignette
  .post_data
  .vignette_footer
  .tags_and_time_to_read
  .time {
  margin-left: 5px;
}
main .authors_intro .allposts .post_vignette .post_image {
  width: 27%;
  background-color: black;
  margin-left: 5px;
}
main .authors_recommendations {
  width: 28%;
  margin-left: 2px;
  background-color: white;
}
