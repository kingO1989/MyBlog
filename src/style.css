.flex-container {
  display: flex;
  flex-direction: row;
}

.item {
  flex: 30%;
}

.flex-post-container {
  display: flex;
  flex-flow: column wrap;
}
.post-title {
  width: 200px;
}
.post-author {
  width: 200px;
  flex-basis: 30px;
}

.post-detail {
  width: 500px;
}

.post-likes {
  width: 100px;
}

.post-comments {
  flex-basis: 30px;
}

@media (max-width: 650px) {
  .flex-container {
    flex-direction: column;
    height: 300px;
  }

  .item {
    flex: 50%;
  }
}
