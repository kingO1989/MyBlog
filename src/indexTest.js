import React from 'react';
import './index.css';
import App from './App';
import { BrowserRouter } from "react-router-dom";
import { DbContextProvider } from "./utilities/DBContext";


export const Main = () => {

    return (<>
        <BrowserRouter>
            <DbContextProvider>
                <App />
            </DbContextProvider>
        </BrowserRouter>

    </>);
}

