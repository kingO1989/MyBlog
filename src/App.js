import { BrowserRouter, Routes, Route } from "react-router-dom";
import Layout from "./Pages/Components/Layout";
import Home from "./Pages/Home";
import PostPage from "./Pages/PostPage";
import { DbContext } from "./utilities/DBContext";
import { client } from "./utilities/ApolloClient";
import { useContext, useState, useEffect } from "react";
import './style.css';
import './App.css';






// Use a custom domain as the supabase URL


///
function App() {

  // console.log("app rendered");
  const dbContext = useContext(DbContext);
  const [posts, setPosts] = useState();
  const [isDark, setIsDark] = useState(false);
  async function fetchPosts() {
    const data = await dbContext.GetAllPostsSummary();

    if (data) {

      setPosts(data)
    }
  }

  useEffect(() => {
    fetchPosts();

  }, [])//run once only on initial render
  return (


    <>
      {
        <input
          type="checkbox"
          checked={isDark}
          onChange={e => setIsDark(e.target.checked)}
        />
      }
      <Routes>

        <Route path="/" element={<Layout />}>
          <Route index element={posts ? <Home posts={posts} theme={isDark} /> : ""} />
          <Route path="post/:postid" element={<PostPage />} />

        </Route>
      </Routes>

    </>




  );
}

export default App;
