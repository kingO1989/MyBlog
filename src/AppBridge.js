
import { BrowserRouter, Routes, Route } from "react-router-dom";

import { useState, useEffect } from 'react';
import './style.css';
import { createClient } from '@supabase/supabase-js'
import { DbContextProvider } from "./utilities/DBContext";
import Layout from "./Pages/Components/Layout";
import Home from "./Pages/Home";
import PostPage from "./Pages/PostPage";
import { ApolloProvider } from "@apollo/client";
import { client } from "./utilities/ApolloClient";



// Use a custom domain as the supabase URL



function AppBridge() {
    console.log(client)

    return (


        <ApolloProvider client={client}>

            <Routes>

                <Route path="/" element={<Layout />}>
                    <Route index element={<Home />} />
                    <Route path="post/:postid" element={<PostPage />} />

                </Route>
            </Routes>


        </ApolloProvider >




    );
}

export default AppBridge;
