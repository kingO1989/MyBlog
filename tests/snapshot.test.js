import TestRenderer from 'react-test-renderer'; // ES6
import { Main } from '../src/index';

it("snap shot test", async () => {
    //create a render of the SelectTimeSeries componen

    const AppRenderer = TestRenderer.create(
        <Main />
    );
    //spy on the fetchData method as it is important to the  SelectTimeSeries component
    // const spy = jest.spyOn(utils, 'fetchData');


    //get the json extract so we can perform snapshots and actions
    let STSJson = AppRenderer.toJSON();


    //create snapshot here
    expect(STSJson).toMatchSnapshot();




})



