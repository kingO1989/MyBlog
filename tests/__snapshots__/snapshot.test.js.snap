// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot test 1`] = `
Array [
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />,
  <header
    data_test="header"
  >
    <div
      className="logo_search"
    >
      <div
        className="logo"
      >
        <a
          data_test="logo"
          href="/"
          onClick={[Function]}
        >
          Logo
        </a>
      </div>
      <div
        className="search"
      >
        <input
          id=""
          name=""
          placeholder="search"
          type="text"
        />
      </div>
    </div>
    <div
      className="my_works_social_media_notifications"
    >
      <span
        className="notify"
      >
        <i
          className="fa-regular fa-bell fontawesome_item"
        />
      </span>
      <span
        className="portfolio"
      >
        <a
          className=""
          href="/"
          onClick={[Function]}
        >
          <i
            className="fa-regular fa-address-card fontawesome_item"
          />
           
        </a>
      </span>
      <span
        class="dropdown profile"
      >
        <a
          href="/"
          onClick={[Function]}
        >
           
          <i
            className="fa-solid fa-user fontawesome_item"
          />
           
        </a>
        <div
          class="dropdown-content"
        >
          <br />
          <button
            onClick={[Function]}
          >
            Google Signin
          </button>
        </div>
      </span>
    </div>
  </header>,
]
`;
